ir/pipeline/block.rs

```diff
@@ -240,6 +259,30 @@ impl BlockContext {
     pub fn get_variable_scopes(&self) -> impl Iterator<Item = (Variable, ScopeId)> + '_ {
         self.variable_declaration.iter().map(|(&var, &scope)| (var, scope))
     }
+
+    pub fn visible_variables(&self, root: ScopeId) -> impl Iterator<Item = Variable> + '_ {
+        self.get_variable_scopes().filter_map(move |(var, scope)| {
+            (scope == ScopeId::INPUT || scope == root || self.is_visible_child(scope, root)).then_some(var)
+        })
+    }
+
+    pub fn variable_status_in_scope(&self, var: Variable, scope: ScopeId) -> VariableStatus {
```

Comment:
Maybe we should call this `VariableLocality` instead of `status`?

```
fn variable_locality_in_scope(var, scope) {
  if declared in this scope: Local
  if declared in a child: Child
  else None
}
```

```
VariableLocality {
  Local,
  Child, 
  None
}
```